src/Service/MemoryInfoHandler.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef MEMORY_INFO_HANDLER_H_
+#define MEMORY_INFO_HANDLER_H_
+
+#include "Memory/MemoryInfoProducer.h"
+#include "ProducerEventProcessor.h"
+#include "capture.pb.h"
+
+namespace orbit_service {
+
+class MemoryInfoHandler : public orbit_memory::MemoryInfoListener {
+ public:
+  explicit MemoryInfoHandler(ProducerEventProcessor* producer_event_processor)
+      : producer_event_processor_{producer_event_processor} {}
```

Comment:
Maybe check that `producer_event_processor` is not `nullptr`?

Reply:
Changed accordingly. Thanks! Please refer to the commit *[fixup] Address other review comments*.